### PR TITLE
fix: Support Open Settings shortcut in the editor title area

### LINKED ISSUE #636: [Formatting Preview] Support Open Settings shortcut in the editor title area
Like the native VS Code setting

### PR DESCRIPTION
**Pull request by CsCherrYY**
fix #636

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -253,6 +259,13 @@
           "group": "9_configuration@20",
           "when": "view == javaProjectExplorer && viewItem =~ /java:project(?=.*?\\b\\+unmanagedFolder\\b)(?=.*?\\b\\+uri\\b)/"
         }
+      ],
+      "editor/title": [
+        {
+          "command": "java.formatterSettings.showTextEditor",
+          "when": "activeCustomEditorId == java.formatterSettingsEditor && !readOnlyMode",
```
Comment (jdneo) [inline package.json:266]: I think it's ok to show the readonly text editor when the setting points to a url

File: package.json [hunk 2]
```diff
@@ -199,6 +199,12 @@
         "command": "java.formatterSettings",
         "category": "Java",
         "title": "Open Java Formatter Settings with Preview"
+      },
+      {
+        "command": "java.formatterSettings.showTextEditor",
```
Comment (jdneo) [inline package.json:204]: disable it in other unnecessary entries, like the command palette

File: src/formatter-settings/index.ts [hunk 3]
```diff
@@ -61,6 +62,14 @@ export class JavaFormatterSettingsEditorProvider implements vscode.CustomTextEdi
         vscode.commands.executeCommand("vscode.openWith", filePath, "java.formatterSettingsEditor");
     }
 
+    public reopenWithTextEditor = instrumentOperation("formatter.showTextEditor", (operationId: string, uri: any) => {
+        sendInfo(operationId, { editor: "RawTextEditor" });
+        if (uri instanceof vscode.Uri) {
+            this.webviewPanel?.dispose();
```
Comment (jdneo) [inline src/formatter-settings/index.ts:68]: What will happen if we do not dispose it?

Comment (CsCherrYY) [reply src/formatter-settings/index.ts:68]: We'll have another raw text editor open, without closing the current custom editor.

File: package.json (hunk 1 above)
Comment (CsCherrYY) [reply package.json:266]: +1 for VSCode will show readonly tips
remote

File: src/formatter-settings/index.ts (hunk 3 above)
Comment (CsCherrYY) [reply src/formatter-settings/index.ts:68]: Update: VSCode behavior is to open another raw editor without closing the current editor, we'd better follow VSCode's.

**Review by jdneo — APPROVED** [review]

**Pull request merged**